src/librustdoc/doctree.rs

```diff
@@ -64,31 +64,34 @@ impl Module<'hir> {
     }
 
     pub(crate) fn push_item(&mut self, new_item: Item<'hir>) {
-        for item_iter in self.items.iter_mut() {
-            if item_iter.name() == new_item.name() {
-                if item_iter.from_glob {
-                    debug!("push_item: {:?} shadowed by {:?}", *item_iter, new_item);
-                    *item_iter = new_item;
+        if let Some(existed_item) =
```

Comment:
nit: the English is a little off.
```suggestion
        if let Some(existing_item) =
```